jupyterhub_moss/utils.py

```diff
@@ -49,24 +47,18 @@ def parse_timelimit(timelimit: str) -> datetime.timedelta:
 def create_prologue(
     default_prologue: str,
     environment_path: str,
-    partition_environments: Iterable[JupyterEnvironment],
+    environment_modules: str,
 ) -> str:
     """Create prologue commands"""
     prologue = default_prologue
 
-    corresponding_default_env = find(
-        lambda env: env.path == environment_path,
-        partition_environments,
-    )
-    if corresponding_default_env is not None:
-        prologue += f"\n{corresponding_default_env.prologue}"
-
+    # Prepend path to environement
     # Singularity images are never added to PATH
-    if environment_path.endswith(".sif"):
-        return prologue
+    if environment_path and not environment_path.endswith(".sif"):
+        prologue += f'\nexport PATH="{environment_path}:$PATH"'
+
+    # Load environment modules
+    if environment_modules:
+        prologue += f"\nmodule load {environment_modules}"
```

Comment:
It's probably better to load modules before updating the `PATH` so the added path is first.

Reply:
good point, custom paths should indeed have precedence over modules